tpmutil/structures.go

```diff
@@ -48,3 +50,11 @@ type responseHeader struct {
 
 // A Handle is a reference to a TPM object.
 type Handle uint32
+
+// TPMSelfMarshaler is used to support parsing new TPM structs without adding special
```

Comment:
Describe primarily what the type represents, not what the motivation behind it is.
`TPMSelfMarshaler allows custom types to override default encoding/decoding behavior in Pack, Unpack and UnpackBuf.`